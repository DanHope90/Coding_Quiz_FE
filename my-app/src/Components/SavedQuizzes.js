import React from "react";
import PropTypes from "prop-types";
import SavedQuizCard from "./SavedQuizCard";
import data from "../Data/SavedQuizzes.json";

function SavedQuizzes() {
  const { quizzes } = data;
  return (
    <>
      <div>This is the Saved Quizzes Page</div>
      <div>
        {quizzes.map((quiz) => (
          <div key={quiz.quizName}>
            <SavedQuizCard quizName={quiz.quizName} quizScore={quiz.quizScore} />
          </div>
        ))}
      </div>
    </>
  );
}

export default SavedQuizzes;

SavedQuizzes.propTypes = {
  quizName: PropTypes.string,
  quizScore: PropTypes.number,
}.isRequired;
