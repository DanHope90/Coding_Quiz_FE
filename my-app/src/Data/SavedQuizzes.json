{
    "quizzes": [
        {
            "quizName": "JavaScript",
            "quizScore": 6
        },
        {
            "quizName": "CSS",
            "quizScore": 4
        },
        {
            "quizName": "HTML",
            "quizScore": 6
        },
        {
            "quizName": "Python",
            "quizScore": 9
        },
        {
            "quizName": "React",
            "quizScore": 2
        },
        {
            "quizName": "Ruby",
            "quizScore": 3
        },
        {
            "quizName": "R",
            "quizScore": 10
        },
        {
            "quizName": "C#",
            "quizScore": 1
        },
        {
            "quizName": "C++",
            "quizScore": 2
        }
    ]
}