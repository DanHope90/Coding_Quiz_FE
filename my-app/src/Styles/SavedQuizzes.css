.saved-quizzes{
    display: flex;
    flex-flow: wrap;
    text-align: center;
}

.indiv-quiz{
    border: 2px solid black;
    padding: 15px;
    margin: 10px;
}