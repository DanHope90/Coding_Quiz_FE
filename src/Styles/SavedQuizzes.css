.saved-quizzes{
    display: flex;
    flex-flow: wrap;
    justify-content: center;
    text-align: center;
}

.indiv-quiz{
    border: 2px solid black;
    padding: 15px;
    margin: 10px;
}