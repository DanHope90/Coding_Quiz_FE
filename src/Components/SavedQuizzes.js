import React, { useContext, useState, useEffect } from "react";
import axios from "axios";
import Context from "../Utils/Context";
import "../Styles/SavedQuizzes.css";

function SavedQuizzes() {
  const { userInfo, setAlert } = useContext(Context);

  const [completedQuizzes, setCompletedQuizzes] = useState([]);

  useEffect(() => {
    axios.get(`https://dev-quiz-22.herokuapp.com/api/user/${userInfo.id}`)
      .then(({ data }) => setCompletedQuizzes(data.savedQuizzes))
      .catch(() => setAlert({ message: "Axios promise rejected", isSuccess: false }));
  }, []);

  // eslint-disable-next-line no-console
  console.dir(completedQuizzes);

  return (
    <div className="saved-quiz-body">This is the Saved Quizzes Page</div>
  );
}

export default SavedQuizzes;
