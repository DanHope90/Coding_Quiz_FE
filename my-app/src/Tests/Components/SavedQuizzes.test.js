import SavedQuizzes from "../../Components/SavedQuizzes";
import React from "react";
import renderer from "react-test-renderer";

test("renders the SavedQuizzes component correctly", () => {
    const testRenderer = renderer.create(<SavedQuizzes />);
    expect(testRenderer).toMatchSnapshot();
});
