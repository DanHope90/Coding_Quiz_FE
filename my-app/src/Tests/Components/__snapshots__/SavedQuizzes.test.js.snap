// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SavedQuizzes component correctly 1`] = `
Array [
  <div>
    This is the Saved Quizzes Page
  </div>,
  <div>
    <div>
      <div>
        JavaScript
      </div>
      <div>
        6
         
        / 10
      </div>
    </div>
    <div>
      <div>
        CSS
      </div>
      <div>
        4
         
        / 10
      </div>
    </div>
    <div>
      <div>
        HTML
      </div>
      <div>
        6
         
        / 10
      </div>
    </div>
    <div>
      <div>
        Python
      </div>
      <div>
        9
         
        / 10
      </div>
    </div>
    <div>
      <div>
        React
      </div>
      <div>
        2
         
        / 10
      </div>
    </div>
    <div>
      <div>
        Ruby
      </div>
      <div>
        3
         
        / 10
      </div>
    </div>
    <div>
      <div>
        R
      </div>
      <div>
        10
         
        / 10
      </div>
    </div>
    <div>
      <div>
        C#
      </div>
      <div>
        1
         
        / 10
      </div>
    </div>
    <div>
      <div>
        C++
      </div>
      <div>
        2
         
        / 10
      </div>
    </div>
  </div>,
]
`;
